src/core/qgspointlocator.cpp

```diff
@@ -704,13 +715,39 @@ bool QgsPointLocator::rebuildIndex( int maxFeaturesToIndex )
     }
     request.setFilterRect( rect );
   }
+
+  bool filter = false;
+  std::unique_ptr< QgsFeatureRenderer > renderer( mLayer->renderer() ? mLayer->renderer()->clone() : nullptr );
+  QgsRenderContext *ctx = nullptr;
+  if ( mContext )
+  {
+    mContext->expressionContext() << QgsExpressionContextUtils::layerScope( mLayer );
+    ctx = mContext.get();
+    if ( renderer )
+    {
+      // setup scale for scale dependent visibility (rule based)
+      renderer->startRender( *ctx, mLayer->fields() );
+      filter = renderer->capabilities() & QgsFeatureRenderer::Filter;
```

Comment:
set request's filter expression here to renderer->filter()

Reply:
That's why in this branch you should set the subset of attributes to those required by the renderer (renderer->usedAttributes())

Reply:
oh yes. Thank you!